library/swift/src/stats/Counter.swift

```diff
@@ -1,24 +1,15 @@
-@_implementationOnly import EnvoyEngine
```

Comment:
nit: on android we usually use `I` to prefix an interface (similar to ios protocol). if this was an android class, it would be named `ICounter`.

personally i don't have too strong of an opinion on this, but `Counter` definitely doesn't not intuitively suggest an interface.

Reply:
looking at other protocols on iOS - `StatsClient` being one example - we don't use `I` prefix for protocols on iOS. Instead, we add `Impl` suffix to objects implementing a given protocol.

Reply:
> looking at other protocols on iOS 
im assuming you are referencing to other protocols in this repo (instead of swift in general), my original question was more about swift in general.

Reply:
In general, `I` is not used as prefix for protocol names in Swift. This is what official Swift guidelines say:

<img width="886" alt="Screen Shot 2020-08-27 at 2 49 58 PM" src="https://user-images.githubusercontent.com/4410346/91498459-a4fb3b80-e874-11ea-8b95-18feb4eee1a2.png">

source: https://swift.org/documentation/api-design-guidelines/

Reply:
An `I-` prefix is much more a Java convention. In Swift, if a distinction is necessary, the `-Protocol` suffix would be more typical. Here, we've simply standardized on `-Impl` for the implementation, since it's common enough across languages, and what Envoy itself uses as well.

Reply:
i see, thank you both.